### PR TITLE
Minor fixes for tagging django admin [FC-0036] 

### PR DESCRIPTION
**Pull request by bradenmacdonald**
This fixes a few minor issues I found while reviewing https://github.com/openedx/edx-platform/pull/33563 :

* Some minor improvements to the django admin. I know we don't want to rely on the django admin too much, but it's sometimes handy to inspect what's going on with tagging/taxonomy model data.
* Explicitly remove "Add" button for ObjectTags and Tags.
* Editing a "normal" Taxonomy in the django admin was not possible, because `_taxonomy_class` didn't have `blank=True`. I fixed that.
* Some fields in the admin like `ObjectTag.tag` were loading all possible tags into a select widget, which could be hundreds of thousands. I changed it to an auto-complete widget.
* On my machine, the Language Taxonomy didn't have the right `_taxonomy_class` set. I think this is due to the messy migration history. I included a fix in case anyone else has the same issue on their machine. I was trying to figure out why the API was not showing "Languages" as system-defined.

ObjectTag before:

Screenshot 2023-11-09 at 11 32 42 AM

ObjectTag after:

Screenshot 2023-11-09 at 11 36 26 AM

Tag before:

Screenshot 2023-11-09 at 11 34 12 AM

Tag after:

Screenshot 2023-11-09 at 11 36 05 AM

Private-ref: FAL-3523

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @bradenmacdonald! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Review by pomegranited — APPROVED** [review]
:+1: Thank you for these fixes @bradenmacdonald !

- [x] I tested this using Django Admin on my devstack.
- [x] I read through the code
- [x] <strike>I checked for accessibility issues</strike> N/A
- [x] Includes documentation

File: openedx_tagging/core/tagging/rest_api/v1/views.py [hunk 1]
```diff
@@ -170,7 +170,8 @@ class TaxonomyView(ModelViewSet):
 
     """
 
-    lookup_value_regex = r"\d+"
+    # System taxonomies use negative numbers for their primary keys
+    lookup_value_regex = r'-?\d+'
```
Comment (bradenmacdonald) [inline openedx_tagging/core/tagging/rest_api/v1/views.py:174]: Thanks for this fix - looks good! 👍🏻

**Comment by openedx-webhooks** [comment]
@bradenmacdonald 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-learning&entry.867055334=https://github.com/openedx/openedx-learning/pull/114&entry.1484655318=https://github.com/bradenmacdonald&entry.752974735=2023-11-09+19:43&entry.1917517419=2023-11-10+02:07&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**